### PR TITLE
Distributed Torch Runner

### PR DESCRIPTION
**Pull request by richardliaw**
## What do these changes do?

Introduces a torchrunner for distributed Torch model training. 

 - [ ] Modularize the model, optimizer, and data creator.
 - [ ] Licensing? A lot of this code is from other places (https://pytorch.org/tutorials/beginner/aws_distributed_training_tutorial.html)
 - [ ] Add training hooks

A current MVP is in this PR, and there's a lot of things to do to fix this up.
Logging, proper metrics, etc.

## Linter

- [x] I've run `scripts/format.sh` to lint the changes in this PR.

cc @ericl @pcmoritz @robertnishihara

### CODE REVIEW

File: python/ray/tune/ray_trial_executor.py [hunk 1]
```diff
@@ -101,7 +102,10 @@ def logger_creator(config):
 
         # Logging for trials is handled centrally by TrialRunner, so
         # configure the remote runner to use a noop-logger.
-        return cls.remote(config=trial.config, logger_creator=logger_creator)
+        return cls.remote(
```
Comment (richardliaw) [inline python/ray/tune/ray_trial_executor.py]: I'd like this to go in (the ability to notify the Trainable of its resource allocation) although I'm not sure if this belongs in this PR. 

One thing that this PR currently assumes is that the TorchRunner will automatically create N workers depending on its resource allocation.

cc @ericl

File: python/ray/experimental/torch/helpers.py [hunk 2]
```diff
@@ -0,0 +1,232 @@
+import math
```
Comment (pcmoritz) [inline python/ray/experimental/torch/helpers.py]: Is this file supposed to be names pytorch_helpers? (as used in the below file)

Comment (richardliaw) [reply python/ray/experimental/torch/helpers.py]: addressed

**Comment by richardliaw** [comment]
Closing this for #4797

**Pull request closed without merging**